service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -233,8 +234,16 @@ private DrsObject fetchObjectInfo(
       drsApi.setBearerToken(bearerToken.getToken());
       if (authorizations.stream()
           .anyMatch(a -> a.drsAuthType() == Authorizations.SupportedTypesEnum.PASSPORTAUTH)) {
-        Optional<List<String>> passports = authService.fetchPassports(bearerToken);
-        return drsApi.postObject(Map.of("passports", passports.orElse(List.of(""))), objectId);
+        try {
+          List<String> passports = authService.fetchPassports(bearerToken).orElse(List.of());
+          if (!passports.isEmpty()) {
+            return drsApi.postObject(Map.of("passports", passports), objectId);
```

Comment:
Too bad...I wish that the code gen would make postObject take a typed object. I'mm actually kinda surprised that it doesn't

Reply:
I agree with you… this seems like a similar issue to what we encountered with calling the TPS client in TDR.  I was surprised that it didn't enforce the presence of a required argument.

Reply:
I've been poking a little...It's because (I think) the code gen only creates models for objects in the `components.schemas` part of the openapi yaml and these are in `components.requestBodies`.  Trying to see if there's a way to have it generate models for those too